docs/design/jet/025-jet-job-partition-pruning.md

```diff
@@ -121,6 +123,26 @@ partitions and for all relations. Then, we extract required partitions set to ru
 baked by `__sql.requiredPartitions` argument. This indicates that `ExecutionPlanBuilder` will
 try to choose only the required members.
 
+##### Mixed approach
+
+In addition to SQL-oriented approach, described above, we can attach the DAG analysis in
+`ExecutionPlanBuilder` during required members calculation to quickly determine required members to run the job.
+We are interested on it for two cases:
+
+- support `allToOne` edge : all edges are directed to one member and partitioned by single key.
+  These partitions and members (if multiple) are also a subjects to be included into the DAG result.
+  Main motivation here is that most of the queries highly likely will have `allToOne` to forward
+  scanned data to the coordinator, and we want to calculate it in proper place in less error-prone way.
+
+- general `partitioned` edge support to unlock support of operations like aggregations. DAG analysis result
+  will include boolean flag, which determines if we need preserve all partitions in the cluster.
+
+Based on described above, the result of DAG analysis may influence on partitions assignment: if DAG has
+`distributed-partitioned` edge, non-used partitions will be assigned to all required members to run the job.
+It is a trade-off between potential performance gains and functionality support coverage,
```

Comment:
I don't understand what the tradeoff is. Including all partitions? It is not a tradeoff, it is a consequence of other assumptions in Jet which are worth mentioning here.

Reply:
It's a trade-off, because: OR you don't support operators requiring `distributed-partitioned` edges and **not** assign all partitions across the required members (which leads to performance penalties), OR you don't support these operators, but you include only required partitions on required members.

Reply:
it would be a tradeoff if in each alternative we would be gaining something and loosing something.

Option 1: no `partitioned` edge support, in case when `partitioned` edge belong to the DAG we cannot use pruning and assign all partitions across all cluster members
Option 2: `partitioned` edge support, in case when `partitioned` edge belongs to the DAG we  assign all partitions across the required members
Option 3: `partitioned` edge support, in case when `partitioned` edge belongs to the DAG we change partitioning of items on edges to use only partitionIds belonging to the job.

Option 2 is strictly better that Option 1. The only price is additional analysis code, but I do not consider that as a tradeoff.
Option 3 vs 2 is much more complicated discussion.